src/osdep/osdep.c

```diff
@@ -115,7 +130,9 @@ set_blocking_fd(int fd)
 	 * this is correct usage of winsock
 	 * when compiling for win32 */
 	u_long mode = 0; // set blocking socket
-	return ioctlsocket(fd, FIONBIO, &mode);
+	if (is_socket(fd)) {
```

Comment:
dtto

Reply:
P.S. sorry to add this to the change but without that it's kind of show-stopper on win64